src/main/java/eu/dissco/core/digitalmediaobjectprocessor/repository/DigitalMediaObjectRepository.java

```diff
@@ -5,29 +5,36 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import eu.dissco.core.digitalmediaobjectprocessor.domain.DigitalMediaObject;
+import eu.dissco.core.digitalmediaobjectprocessor.domain.DigitalMediaObjectKey;
 import eu.dissco.core.digitalmediaobjectprocessor.domain.DigitalMediaObjectRecord;
 import java.time.Instant;
-import java.util.Optional;
+import java.util.Collection;
+import java.util.List;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.jooq.DSLContext;
 import org.jooq.JSONB;
+import org.jooq.Query;
 import org.jooq.Record;
 import org.springframework.stereotype.Repository;
 
+@Slf4j
 @Repository
 @RequiredArgsConstructor
 public class DigitalMediaObjectRepository {
 
   private final DSLContext context;
   private final ObjectMapper mapper;
 
-  public Optional<DigitalMediaObjectRecord> getDigitalMediaObject(String digitalSpecimenId,
-      String mediaUrl) {
+  public List<DigitalMediaObjectRecord> getDigitalMediaObject(
+      List<DigitalMediaObjectKey> digitalMediaKeys) {
     return context.select(NEW_DIGITAL_MEDIA_OBJECT.asterisk())
         .from(NEW_DIGITAL_MEDIA_OBJECT)
-        .where(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(digitalSpecimenId))
-        .and(NEW_DIGITAL_MEDIA_OBJECT.MEDIA_URL.eq(mediaUrl))
-        .fetchOptional(this::mapDigitalMediaObject);
+        .where(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.in(
+            digitalMediaKeys.stream().map(DigitalMediaObjectKey::digitalSpecimenId).toList()))
+        .and(NEW_DIGITAL_MEDIA_OBJECT.MEDIA_URL.in(
+            digitalMediaKeys.stream().map(DigitalMediaObjectKey::mediaUrl).toList()))
+        .fetch(this::mapDigitalMediaObject);
```

Comment:
I think it would be more efficient to split DigitalMediaObjectKey in the service and pass two lists into this function. That way you're not doing two streams per object.